### ISSUE TITLE
Inpainting with non-binary masks

### ISSUE THREAD

**Issue by tachella** [deepinv/deepinv#1167]
### Environment

all

### Describe the bug

The Inpainting class generates wrong measurements when the masks are not binary, due to the noise masking [here](https://github.com/deepinv/deepinv/blob/1d4d6f83303632893704ff5b16aac29050585cf2/deepinv/physics/inpainting.py#L133) which should be applied only if the mask is *exacty* zero.

### Steps to reproduce

```
import deepinv as dinv
import torch

device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

img_size = (1, 3, 32, 32)

mask = torch.ones(img_size, device=device)
mask[..., :16] = 0.001

physics = dinv.physics.Inpainting(mask=mask, device=device, img_size=img_size[1:])

x = torch.ones(img_size, device=device)

y = physics(x)

x_dagger = physics.A_dagger(y)

assert x_dagger==x
```

(should be true, is false)

### Expected behavior

The mask should be applied to the noise only if it is exactly zero.
